srcpkgs/libmtp/template

```diff
@@ -1,17 +1,21 @@
 # Template file for 'libmtp'
 pkgname=libmtp
-version=1.1.18
+version=1.1.19
 revision=1
 build_style=gnu-configure
-configure_args="--disable-static --with-udev=/usr/lib/udev"
+configure_args="--disable-static --with-udev=/usr/lib/udev
+ --with-udev-group=_libmtp"
```

Comment:
Was it using a default group before this? Or none?

Reply:
The default is "none" which means there's no group. The only change with this seems to be in the file `/usr/lib/udev/rules.d/69-libmtp.rules`:
```diff
--- 69-libmtp.rules.old	2020-11-05 23:30:09.000000000 -0300
+++ 69-libmtp.rules.new	2021-12-29 18:26:43.895937081 -0300
@@ -9,7 +9,7 @@
 LABEL="libmtp_usb_rules"
 
 # If we have a hwdb entry for this device, act immediately!
-ENV{ID_MTP_DEVICE}=="1", SYMLINK+="libmtp-%k", GOTO="libmtp_rules_end"
+ENV{ID_MTP_DEVICE}=="1", SYMLINK+="libmtp-%k", GROUP="_libmtp", GOTO="libmtp_rules_end"
 
 # Fall back to probing.
 # Some sensitive devices we surely don't wanna probe
@@ -36,6 +36,6 @@
 ENV{ID_USB_INTERFACES}=="*:0701??:*", GOTO="libmtp_rules_end"
 
 # Autoprobe vendor-specific, communication and PTP devices
-ENV{ID_MTP_DEVICE}!="1", ENV{MTP_NO_PROBE}!="1", ENV{COLOR_MEASUREMENT_DEVICE}!="1", ENV{ID_GPHOTO}!="1", ENV{libsane_matched}!="yes", ATTR{bDeviceClass}=="00|02|06|ef|ff", PROGRAM="/usr/lib/udev/mtp-probe /sys$env{DEVPATH} $attr{busnum} $attr{devnum}", RESULT=="1", SYMLINK+="libmtp-%k", ENV{ID_MTP_DEVICE}="1", ENV{ID_MEDIA_PLAYER}="1"
+ENV{ID_MTP_DEVICE}!="1", ENV{MTP_NO_PROBE}!="1", ENV{COLOR_MEASUREMENT_DEVICE}!="1", ENV{ID_GPHOTO}!="1", ENV{libsane_matched}!="yes", ATTR{bDeviceClass}=="00|02|06|ef|ff", PROGRAM="/usr/lib/udev/mtp-probe /sys$env{DEVPATH} $attr{busnum} $attr{devnum}", RESULT=="1", SYMLINK+="libmtp-%k", ENV{ID_MTP_DEVICE}="1", ENV{ID_MEDIA_PLAYER}="1", GROUP="_libmtp"
 
 LABEL="libmtp_rules_end"
```

The net effect is that the device node that is created in `/dev/bus/usb/...` has its group set to `_libmtp`. These files are created with mode 664 in any case, but with group `root`.

Reply:
ping @ericonr

Besides the update, this is solving an issue: libmtp doesn't work out-of-the-box unless it is run on sudo, and I don't want to run gtk apps on sudo. E.g. I'm using gmtp: with this patch and me on the _libmtp group it works fine, otherwise it cannot access the device.

Reply:
neither Alpine, Arch nor Debian don't set this...?

Reply:
maybe a different group makes sense?

Reply:
> maybe a different group makes sense?

Maybe audio?

Reply:
Gentoo uses `plugdev`: https://gitweb.gentoo.org/repo/gentoo.git/tree/media-libs/libmtp/libmtp-1.1.19.ebuild#n58

Arch documentation claims to use `uucp` but I don't think the pkgbuild actually sets that: https://wiki.archlinux.org/title/Media_Transfer_Protocol#libmtp

Debian I don't know.

Reply:
I checked their build templates, all of them don't set it

Reply:
> I checked their build templates, all of them don't set it

So, how do they solve the problem I described above:
> libmtp doesn't work out-of-the-box unless it is run on sudo, and I don't want to run gtk apps on sudo. E.g. I'm using gmtp: with this patch and me on the _libmtp group it works fine, otherwise it cannot access the device.

What's the downside of setting this option?

Reply:
so  libmtp is broken on all the other distros then?
i would  go with what gentoo does (`plugdev`), as that's used in some templates already

Reply:
> so libmtp is broken on all the other distros then?

I don't know, I don't use other distros. Is there a different method to give users access to usb devices? The only methods that come to my mind are: (a) make the device group-readable by a well known group (b) run the programs that need to access the device as root (either setuid or sudo)  (c) have a daemon running as root that the programs talk to. I don't think libmtp has (c) and (b) seems unsafe so (a) is the only reasonable choice AFAICT.

> i would go with what gentoo does (`plugdev`), as that's used in some templates already

- `plugdev` is not documented here: https://docs.voidlinux.org/config/users-and-groups.html#default-groups
- I see three packages create group `plugdev`, namely `openocd`, `openrazer-meta` and `wally-udev-rules`, so maybe it's ok to use `plugdev` without underscore (the policy says to use underscore: "System accounts and groups must be prefixed with an underscore to prevent clashing with names of user accounts." except for old packages).
- debian says `plugdev`  "Allows members to mount (only with the options nodev and nosuid, for security reasons) and umount removable devices through pmount." (https://wiki.debian.org/SystemGroups)
- in void pmount is setuid root instead.

My rationale for `_libmtp` was that it is unambiguous what you are getting access to.

Reply:
After this discussion and a chat with @0x5c on irc, it seems using `plugdev` is the better way.